### ISSUE TITLE
NCL loss not decrease - Validation accuracy in seen dataset not increase

### ISSUE THREAD

**Issue by xthyax** [zhunzhong07/NCL#7]
Hi, thank you for your great work!

I've been following your repo to try to make it work on my own private dataset. In the self-supervised and supervised stage, everything works as expected, however, when shifting into the NCL stage, the training loss seems not to decrease, and the Validation accuracy on the seen dataset does not increase, but on the other hand, Validation Accuracy on the unseen dataset is peak 1.0 almost all the time. I'm not sure is it because the model was not learning or I'm missing something.

![Screenshot 2023-10-05 144304](https://github.com/zhunzhong07/NCL/assets/43241702/f58d1ef3-361b-4e3b-893d-bdb1ae9393b6)
![image](https://github.com/zhunzhong07/NCL/assets/43241702/d280a4d6-0484-42ab-9f75-30cd9df1848c)
